addons/account/models/account.py

```diff
@@ -176,8 +176,7 @@ def name_search(self, name, args=None, operator='ilike', limit=100):
 
     @api.onchange('internal_type')
     def onchange_internal_type(self):
-        if self.internal_type in ('receivable', 'payable'):
```

Comment:
this part isn't correct:
* changing the internal type to 'receivable' or 'payable' have to set the reconcile field to True
* changing the internal type to something else have to keep the reconcile field untouched